terraform/aws/modules/elasticache/elasticache.tf

```diff
@@ -0,0 +1,122 @@
+# Create memcached cache cluster
+
+resource "aws_elasticache_cluster" "memcached" {
+    cluster_id                     = "${var.environment}-${var.name}"
+    engine                         = var.engine
+    engine_version                 = var.engine_version
+    node_type                      = var.instance_type
+    num_cache_nodes                = var.cluster_size
+    parameter_group_name           = aws_elasticache_parameter_group.cache_parameter_group.id
+    subnet_group_name              = "${aws_elasticache_subnet_group.cache_subnet_group.id}"
+    security_group_ids             = [aws_security_group.cache_sg.id]
+    maintenance_window             = var.maintenance_window
+    apply_immediately              = var.apply_immediately
+    notification_topic_arn         = var.notification_topic_arn
+    port                           = var.port
+    az_mode                        = var.az_mode
+    availability_zone              = var.availability_zone
+    preferred_availability_zones   = var.availability_zones
+    count                          = var.engine == "memcached" ? 1 : 0
+}
+
+# Create Redis cache single node cluster
+
+resource "aws_elasticache_cluster" "redis_node" {
+    cluster_id                     = "${var.environment}-${var.name}-node"
+    engine                         = var.engine
+    engine_version                 = var.redis_version
+    node_type                      = var.instance_type
+    num_cache_nodes                = var.cluster_size
+    parameter_group_name           = aws_elasticache_parameter_group.cache_parameter_group.id
+    subnet_group_name              = "${aws_elasticache_subnet_group.cache_subnet_group.id}"
+    security_group_ids             = [aws_security_group.cache_sg.id]
+    maintenance_window             = var.maintenance_window
+    apply_immediately              = var.apply_immediately
+    notification_topic_arn         = var.notification_topic_arn
+    port                           = var.port
+    az_mode                        = var.az_mode
+    availability_zone              = var.availability_zone
+    preferred_availability_zones   = var.availability_zones
+    count                          = var.engine == "redis" && var.replication_group_enabled == "disabled" ? 1 : 0
```

Comment:
Can you please move the count to the top of the resource. It keeps it more readable.

Reply:
Done, please check.